src/test/java/io/odpf/depot/http/HttpSinkTest.java

```diff
@@ -124,7 +128,49 @@ public void shouldReportErrors() throws IOException {
         when(httpSinkClient.send(validRecords)).thenReturn(responses);
         when(httpSinkClient.send(records)).thenReturn(responses);
 
-        HttpSink httpSink = new HttpSink(httpSinkClient, request, instrumentation);
+        Map<Integer, Boolean> retryStatusCodeRanges = new HashMap<>();
+
+        HttpSink httpSink = new HttpSink(httpSinkClient, request, retryStatusCodeRanges, instrumentation);
+        OdpfSinkResponse odpfSinkResponse = httpSink.pushToSink(messages);
+        Assert.assertTrue(odpfSinkResponse.hasErrors());
+        Assert.assertEquals(5, odpfSinkResponse.getErrors().size());
+        Assert.assertEquals(ErrorType.SINK_5XX_ERROR, odpfSinkResponse.getErrorsFor(1).getErrorType());
+        Assert.assertEquals(ErrorType.SINK_5XX_ERROR, odpfSinkResponse.getErrorsFor(3).getErrorType());
+        Assert.assertEquals(ErrorType.SINK_5XX_ERROR, odpfSinkResponse.getErrorsFor(4).getErrorType());
+    }
+
+    @Test
+    public void shouldReportErrorsBasedOnStatusCodeWhenStatusCodeDoesNotFallUnderConfiguredRetryStatusCodeRange() throws IOException {
```

Comment:
i dont think we need add more test in this class for this particular case. this particular case should be handled in response parser test.